explainaboard/loaders/kg_link_tail_prediction.py

```diff
@@ -68,56 +70,35 @@ def load(self) -> list[dict]:
-                }
-
-                # additional user-defined features
+        for features_dict in raw_data.raw_data:
+            data_i = {
+                "id": features_dict["example_id"],  # should be string type
+                "true_head": entity_dic[features_dict["gold_head"]]["label"]
+                if features_dict["gold_head"] in entity_dic.keys()
+                else features_dict["gold_head"],
+                "true_link": features_dict["gold_predicate"],
+                "true_tail": entity_dic[features_dict["gold_tail"]]["label"]
+                if features_dict["gold_tail"] in entity_dic.keys()
+                else features_dict["gold_tail"],
+                "true_tail_anonymity": features_dict["gold_tail"],
+                "true_label": features_dict[
+                    "gold_" + features_dict["predict"]
+                ],  # the entity to which we compare the predictions
+                "predictions": features_dict["predictions"],
+                "true_rank": features_dict.get(
+                    "true_rank", None
+                ),  # rank of the true entity in predictions
+            }
+
+            # additional user-defined features
+            if raw_data.metadata.custom_features:
                 data_i.update(
                     {
                         feature_name: features_dict[feature_name]
-                        for feature_name in self.user_defined_features_configs
+                        for feature_name in raw_data.metadata.custom_features
                     }
                 )
 
-                data.append(data_i)
-        else:
-            for _, (example_id, features_dict) in enumerate(raw_data.items()):
-                data.append(
-                    {
-                        "id": str(example_id),  # should be string type
-                        "true_head": entity_dic[features_dict["gold_head"]]["label"]
-                        if features_dict["gold_head"] in entity_dic.keys()
-                        else features_dict["gold_head"],
-                        "true_link": features_dict["gold_predicate"],
-                        "true_tail": entity_dic[features_dict["gold_tail"]]["label"]
-                        if features_dict["gold_tail"] in entity_dic.keys()
-                        else features_dict["gold_tail"],
-                        "true_tail_anonymity": features_dict["gold_tail"],
-                        "true_label": features_dict[
-                            "gold_" + features_dict["predict"]
-                        ],  # the entity to which we compare the predictions
-                        "predictions": features_dict["predictions"],
-                        "true_rank": features_dict.get(
-                            "true_rank", None
-                        ),  # rank of the true entity in predictions
-                    }
-                )
-        return data
+            data.append(data_i)
```

Comment:
I remembered the above code should have been updated in this PR: https://github.com/neulab/ExplainaBoard/blob/5c423ddd3e21fb14e23cfc9122c832e7732923da/explainaboard/loaders/kg_link_tail_prediction.py#L65
It's a little strange to see the outdated code has been updated again.

Reply:
got the answer from the later commits.